exercises/part3/optuna/source.py

```diff
@@ -0,0 +1,101 @@
+CUDA_DEVICE = 0  # Set device ID if you can use GPU.
```

Comment:
I added the runnable example of optimization.
However, I'm wondering if I should put the link to colab because of the following reasons:

1. Require `optuna`. `binder/requirements.txt` seems to keep small.
2. Require GPU. Hyperparameter optimization takes a longer time than other runnable examples.

I want some feedback about it.

Reply:
If it's not feasible to actually run the code in the example, having it be an executable example isn't really helpful.  So maybe a link to colab is better?

Reply:
> If it's not feasible to actually run the code in the example, having it be an executable example isn't really helpful.

Agreed. Let me add links to colab this time. eb32f25